pallets/asset-manager/src/lib.rs

```diff
@@ -179,54 +282,75 @@ pub mod pallet {
 	#[pallet::storage]
 	#[pallet::getter(fn asset_type_units_per_second)]
 	pub type AssetTypeUnitsPerSecond<T: Config> =
-		StorageMap<_, Blake2_128Concat, T::AssetType, u128>;
+		StorageMap<_, Blake2_128Concat, T::ForeignAssetType, u128>;
+
+	/// Stores the counter of the number of local assets that have been
+	/// created so far
+	#[pallet::storage]
+	#[pallet::getter(fn local_asset_counter)]
+	pub type LocalAssetCounter<T: Config> = StorageValue<_, u128, ValueQuery>;
+
+	/// Local asset deposits, a mapping from assetId to a struct
+	/// holding the creator (from which the deposit was reserved) and
+	/// the deposit amount
+	#[pallet::storage]
+	#[pallet::getter(fn local_asset_deposit)]
+	pub type LocalAssetDeposit<T: Config> =
```

Comment:
It is also worth noting that the key is assetId, which is a hash of local counter. Therefore one cannot manipulate the tree at choice but will always depend on the value of the hash

Reply:
I think this should be sufficient to believe that imbalancing attack is somehow infeasible (unless you start creating/destroying assets, which would become incredible expensive)

Reply:
In that case, we could use twox instead of blake2 and it would make using these storage items a bit cheaper.

We've already deployed the original storage with blake2, though, so maybe we should leave it as is and revisit later. Up to you.

Reply:
Makes sense yeah. I will change it, do I need to run benchmark again?

Reply:
Chagned for Twox64, for now I will leave the benchmark as it is, once bench-bot is here I will decrease its value